eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/ArrayListIterate.java

```diff
@@ -1338,6 +1339,27 @@ private ArrayListIterate()
         return RandomAccessListIterate.partitionWith(list, predicate, parameter);
     }
 
+    /**
+     * @see Iterate#indexOf(Iterable, Object)
+     */
+    public static <T> int indexOf(ArrayList<T> list, Object object)
+    {
+        int size = list.size();
+        if (ArrayListIterate.isOptimizableArrayList(list, size))
+        {
+            T[] elements = ArrayListIterate.getInternalArray(list);
+            for (int i = 0; i < size; i++)
+            {
+                if (Objects.equals(elements[i], object))
+                {
+                    return i;
+                }
+            }
+            return -1;
+        }
+        return RandomAccessListIterate.indexOf(list, object);
+    }
```

Comment:
Is this an unrelated change?

Reply:
since I introduced `Iterate.indexOf` needed by `ImmutableSortedSetFromBagAdapter`, I decided to be coherent and add it for the various iterate classes.

Reply:
note it is in a separate commit